src/partition/repart.c

```diff
@@ -1897,6 +1897,37 @@ static int config_parse_encrypted_volume(
 static DEFINE_CONFIG_PARSE_ENUM_WITH_DEFAULT(config_parse_verity, verity_mode, VerityMode, VERITY_OFF, "Invalid verity mode");
 static DEFINE_CONFIG_PARSE_ENUM_WITH_DEFAULT(config_parse_minimize, minimize_mode, MinimizeMode, MINIMIZE_OFF, "Invalid minimize mode");
 
+static int partition_finalize_fstype(Partition *p, const char *path) {
+        int r;
+
+        for (PartitionDesignator d = 0; d < _PARTITION_DESIGNATOR_MAX; d++) {
+                _cleanup_free_ char *e = NULL, *upper = NULL;
+
+                if (!gpt_partition_type_has_filesystem(p->type) || p->type.designator != d)
+                        continue;
+
+                upper = strdup(partition_designator_to_string(d));
+                if (!upper)
+                        return log_oom();
+
+                e = strjoin("SYSTEMD_REPART_OVERRIDE_", ascii_strupper(upper), "_FSTYPE");
```

Comment:
please make thus a suffix thing

Reply:
some deignator strings contain a "-". you need to string_replace_char() that into "_"